packages/round-manager/src/features/round/FundContract.tsx

```diff
@@ -116,7 +116,10 @@ export default function FundContract(props: {
     data: balanceData,
     isError: isBalanceError,
     isLoading: isBalanceLoading,
-  } = useBalance(tokenDetail);
+  } = useBalance({
+    address: tokenDetail.addressOrName as Address,
+    token: tokenDetail.token,
+  });
```

Comment:
What is the difference between the old and new code? They seem equivalent.

Also, I realize it's like this in several areas already, but this would be safer:

```ts
    address: assertAddress(tokenDetail.addressOrName),
    token: tokenDetail.token,
  });

// ...


function assertAddress(value: unknown): Address {
  if (typeof value === "string" && value.slice(0, 2) === "0x") {
    return value as Address;
  } else {
    throw new Error(`Value is not an address (${value})`);
  }
}
```

Reply:
> What is the difference between the old and new code? They seem equivalent.

the new wagmi version needs `{ address: ...., token: ...}`, the previous one had `addressOrName` instead of `address`

Reply:
Duh, apparently I'm code-blind today. Then I guess they could be renamed in the `tokenDetail` and `tokenDetailUser` objects themselves, since they're not used elsewhere, or they could be got rid of entirely?

Reply:
I also missed a few things, I thought the object was created in other places but most of them are just created for that call, I'm checking them all now

Reply:
@bard I updated it using your function

Reply:
> @bard good point I'm going to check it. would you still add the assert in that case?

Yes, I would use the runtime assert in place of every `as Address` we have.